charts/vector/values-agent.yaml

```diff
@@ -0,0 +1,651 @@
+# for Vector Pods.
+dnsPolicy: ClusterFirst
+
+# dnsConfig -- Specify the [dnsConfig](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config)
+# options for Vector Pods.
+dnsConfig: {}
+  # nameservers:
+  #   - 1.2.3.4
+  # searches:
+  #   - ns1.svc.cluster-domain.example
+  #   - my.dns.search.suffix
+  # options:
+  #   - name: ndots
+  #     value: "2"
+  #   - name: edns0
+
+# shareProcessNamespace -- Specify the [shareProcessNamespace](https://kubernetes.io/docs/tasks/configure-pod-container/share-process-namespace/)
+# options for Vector Pods.
+shareProcessNamespace: false
+
+# livenessProbe -- Override default liveness probe settings. If customConfig is used, requires customConfig.api.enabled
+# to be set to true.
+livenessProbe: {}
+  # httpGet:
+  #   path: /health
+  #   port: api
+
+# readinessProbe -- Override default readiness probe settings. If customConfig is used,
+# requires customConfig.api.enabled to be set to true.
+readinessProbe: {}
+  # httpGet:
+  #   path: /health
+  #   port: api
+
+# Configure a PodMonitor for Vector, requires the PodMonitor CRD to be installed.
+podMonitor:
+  # podMonitor.enabled -- If true, create a PodMonitor for Vector.
+  enabled: false
+  # podMonitor.jobLabel -- Override the label to retrieve the job name from.
+  jobLabel: app.kubernetes.io/name
+  # podMonitor.port -- Override the port to scrape.
+  port: prom-exporter
+  # podMonitor.path -- Override the path to scrape.
+  path: /metrics
+  # podMonitor.interval -- Override the interval at which metrics should be scraped.
+  interval:
+  # podMonitor.scrapeTimeout -- Override the timeout after which the scrape is ended.
+  scrapeTimeout:
+  # podMonitor.relabelings -- [RelabelConfigs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config)
+  # to apply to samples before scraping.
+  relabelings: []
+  # podMonitor.metricRelabelings -- [MetricRelabelConfigs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#metric_relabel_configs)
+  # to apply to samples before ingestion.
+  metricRelabelings: []
+  # podMonitor.podTargetLabels -- [podTargetLabels](https://prometheus-operator.dev/docs/operator/api/#monitoring.coreos.com/v1.PodMonitorSpec)
+  # transfers labels on the Kubernetes Pod onto the target.
+  podTargetLabels: []
+  # podMonitor.additionalLabels -- Adds additional labels to the PodMonitor.
+  additionalLabels: {}
+  # podMonitor.honorLabels -- If true, honor_labels is set to true in the [scrape config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config).
+  honorLabels: false
+  # podMonitor.honorTimestamps -- If true, honor_timestamps is set to true in the [scrape config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config).
+  honorTimestamps: true
+
+# Log level for Vector.
+logLevel: "info"
+
+# Optional built-in HAProxy load balancer.
```

Comment:
haproxy is not necessary for agent: https://github.com/WhizardTelemetry/vector-helm-charts/pull/2/files#diff-033177f7a15cb8a53378f2a323022c089d4e39f4601c9469a6b25457436c4b16R456-R623

Reply:
agent is false

Reply:
we can remove these config because they'll never be applied for agent

Reply:
It is kept for comparison with upstream modifications.

Reply:
<img width="656" alt="image" src="https://github.com/WhizardTelemetry/vector-helm-charts/assets/35139861/897cc7f1-6064-4f9d-ba4c-816acec22c12">